### PR TITLE
Index compression

### PR DESCRIPTION
**Pull request by SebastianSchmidl**
depends on PR #49 (please merge that one first)

- add delta encoding to CID lists in posting lists
- do not save dictionary as stringifyed JSON object!
  - maybe use pickle or similar
- compress tf floating point numbers in posting lists (how?)
- switch to binary files (maybe in a new PR?)

### CODE REVIEW

**Comment by SebastianSchmidl** [comment]
TODO:
- adopt SearchEngine to comply with the new interface and design of the Index
- test index on a large dataset (indexing and querying)

**Comment by SebastianSchmidl** [comment]
query results are very bad now...don't know why

File: cse/SearchEngine.py [hunk 1]
```diff
@@ -330,8 +366,10 @@ def processAll(self, tokenTuples):
 
 if __name__ == '__main__':
     se = SearchEngine("data")
-    #se.index()
+    se.index()
```
Comment (Benedikt1992) [inline cse/SearchEngine.py:369]: Really every time?

Comment (Benedikt1992) [reply cse/SearchEngine.py:369]: PR #54

**Review by Benedikt1992 — APPROVED** [review]

File: cse/SearchEngine.py (hunk 1 above)
Comment (SebastianSchmidl) [reply cse/SearchEngine.py:369]: this is only for testing anyway... we have no entry point for the search engine specified yet

it changes from commit to commit 😆

**Pull request merged**